iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Branch.java

```diff
@@ -210,33 +210,46 @@ default ApparentPowerLimits getApparentPowerLimits(Branch.Side side) {
         throw new AssertionError("Unexpected side: " + side);
     }
 
+    default LoadingLimits getLimits(LimitType type, Branch.Side side) {
+        switch (type) {
+            case CURRENT:
+                return getCurrentLimits(side);
+            case ACTIVE_POWER:
+                return getActivePowerLimits(side);
+            case APPARENT_POWER:
+                return getApparentPowerLimits(side);
+            default:
+                throw new UnsupportedOperationException(String.format("Getting %s limits is not supported.", type.name()));
+        }
+    }
+
     boolean isOverloaded();
 
     boolean isOverloaded(float limitReduction);
 
     int getOverloadDuration();
 
-    boolean checkPermanentLimit(Side side, float limitReduction);
+    boolean checkPermanentLimit(Side side, float limitReduction, LimitType type);
 
-    boolean checkPermanentLimit(Side side);
+    boolean checkPermanentLimit(Side side, LimitType type);
 
-    boolean checkPermanentLimit1(float limitReduction);
+    boolean checkPermanentLimit1(float limitReduction, LimitType type);
 
-    boolean checkPermanentLimit1();
+    boolean checkPermanentLimit1(LimitType type);
 
-    boolean checkPermanentLimit2(float limitReduction);
+    boolean checkPermanentLimit2(float limitReduction, LimitType type);
 
-    boolean checkPermanentLimit2();
+    boolean checkPermanentLimit2(LimitType type);
 
-    Overload checkTemporaryLimits(Side side, float limitReduction);
+    Overload checkTemporaryLimits(Side side, float limitReduction, LimitType type);
 
-    Overload checkTemporaryLimits(Side side);
+    Overload checkTemporaryLimits(Side side, LimitType type);
 
-    Overload checkTemporaryLimits1(float limitReduction);
+    Overload checkTemporaryLimits1(float limitReduction, LimitType type);
 
-    Overload checkTemporaryLimits1();
+    Overload checkTemporaryLimits1(LimitType type);
 
-    Overload checkTemporaryLimits2(float limitReduction);
+    Overload checkTemporaryLimits2(float limitReduction, LimitType type);
 
-    Overload checkTemporaryLimits2();
```

Comment:
Do you want it to be a breaking change or can you keep these methods as deprecated? I would leave them as deprecated and returning `CURRENT` type limits in the default implementation (deprecated since 4.3.0)

Reply:
Yes I'll add them back, make them check current, and make them deprecated.